esp/src/eclwatch/HPCCPlatformWidget.js

```diff
@@ -440,15 +451,56 @@ define([
         _onAboutClose: function (evt) {
             this.aboutDialog.hide();
         },
+
+        _onLock: function (evt) {
+            var context = this;
+            
+            xhr("esp/lock", {
+                method: "post",
+            }).then(function(data){
+                context.unlockDialog.show();
+                context.unlockUserName.set("value", context.userName);
+                domClass.add("SessionLock", "overlay");
+            });
+        },
+
+        _onUnlock: function (evt) {
+            var context = this;
+
+            if (this.unlockForm.validate()) {
+                WsAccount.Unlock({
+                    request: {
+                        username: this.unlockUserName.get("value"),
+                        password: this.unlockPassword.get("value")
+                    }
+                }).then(function (response) {
+                    var status = dom.byId("UnlockStatus");
+
+                    if (response.UnlockResponse.Error === 0) {
+                        if (status.innerHTML !== "") {
+                            status.innerHTML = "";
+                        }
+                        context.unlockDialog.hide();
+                        domClass.remove("SessionLock", "overlay");
+                    } else {
+                        status.innerHTML = response.UnlockResponse.Message
+                    }
+                });
+            }
+        },
 		
         _onLogout: function (evt) {
-            xhr("esp/logout", {
-                method: "post"
-            }).then(function(data){
-                if (data){
-                    document.cookie = "ESPSessionID" + location.port + " = '' "; "expires=Thu, 01 Jan 1970 00:00:00 GMT"; // or -1
-                    window.location.reload();
-                }
+            this.logoutConfirm.show();
+            query(".dijitDialogUnderlay").style("opacity", "0.5");
+            this.logoutConfirm.on("execute", function(){
+                xhr("esp/logout",{
+                    method: "post"
+                }).then(function(data){
+                    if (data){
+                        document.cookie = "ESPSessionID" + location.port + " = '' "; "expires=Thu, 01 Jan 1970 00:00:00 GMT"; // or -1
```

Comment:
Should probably use dojo/cookie?  I am not sure what functionality it adds, probably some better cross browser testing at least?